src/main/c/cext/printf.c

```diff
@@ -35,44 +61,37 @@ static int rb_tr_fprintf_value_arginfo(const struct printf_info *info,
   }
   return 1;
 }
-#endif
+#endif  // __APPLE__
 
 static int rb_tr_fprintf_value(FILE *stream,
                                const struct printf_info *info,
                                const void *const *args) {
-  char *cstr = NULL;
-  VALUE v;
-  int len;
+  char *cstr = rb_value_to_str((const VALUE *) args[0], info->showsign);
+  return fprintf(stream, "%s", cstr);
+}
 
-  v = *((const VALUE *) (args[0]));
-  if (info->showsign) {
-    if (RB_TYPE_P(v, T_CLASS)) {
-      if (v == rb_cNilClass) {
-        cstr = "nil";
-      } else if (v == rb_cTrueClass) {
-        cstr = "true";
-      } else if (v == rb_cFalseClass) {
-        cstr = "false";
-      }
-    }
-    if (cstr == NULL) {
-      VALUE str = rb_inspect(v);
-      len = rb_str_len(str);
-      cstr = RSTRING_PTR(str);
-    }
-  } else {
-    VALUE str = rb_obj_as_string(v);
-    len = rb_str_len(str);
-    cstr = RSTRING_PTR(str);
-  }
-  len = fprintf(stream, "%s", cstr);
-  return len;
+void rb_tr_init_printf(void) {
```

Comment:
Could you move this outside the ifdefs and have a single declaration, with the 3 cases inside for clarity?